packages/data-context/src/actions/AppActions.ts

```diff
@@ -1,24 +1,43 @@
+import type Bluebird from 'bluebird'
 import type { FoundBrowser } from '@packages/types'
 import pDefer from 'p-defer'
 
 import type { DataContext } from '..'
 
 export interface AppApiShape {
   getBrowsers(): Promise<FoundBrowser[]>
+  ensureAndGetByNameOrPath(nameOrPath: string, returnAll?: boolean, browsers?: FoundBrowser[]): Bluebird<FoundBrowser | FoundBrowser[] | undefined>
 }
 
 export class AppActions {
   constructor (private ctx: DataContext) {}
 
-  async setActiveBrowser (id: string) {
+  setActiveBrowser (browser?: FoundBrowser) {
```

Comment:
Instead of making this nullable I wonder if it would be more clearer to just make the argument required and then do the check in the other function?

```ts
setActiveBrowser (browser: FoundBrowser) {
  this.ctx.coreData.wizard.chosenBrowser = browser
}

 setActiveBrowserById (id: string) {
    const browserId = this.ctx.fromId(id, 'Browser')

    // Ensure that this is a valid ID to set
    const browser = this.ctx.appData.browsers?.find((b) => this.idForBrowser(b) === browserId)

    if (!browser) {
      return
    }

    this.setActiveBrowser(browser)
  }
````

Since I don't see any use case where we want to set the active browser to be undefined.

Reply:
I guess I was opting to make it DRYer, since that check will have to go in both functions that call this.